### PR TITLE
Log all track seeks

### PR DESCRIPTION
**Pull request by peppy**
I want this in the release to help with https://github.com/ppy/osu/issues/26879

Will probably need some tidying up, but PRing for visibility.

This should be seen as a temporary PR which can be undone later. Alternative is framework changes, which I'd like to avoid.

### CODE REVIEW

**Comment by smoogipoo** [comment]
I don't think it's good that this PR is bundling a bunch of miscellaneous changes - mostly the change from `Track` to `ITrack`. It increases the chance for this PR to not revert cleanly.

File: osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs [hunk 1]
```diff
@@ -192,7 +192,7 @@ private void showFromTime(double time, bool animated)
         private void regenerateDisplay(bool animated)
         {
             // Before a track is loaded, it won't have a valid length, which will break things.
-            if (!beatmap.Value.Track.IsLoaded)
+            if (beatmap.Value.Track.IsLoaded())
```
Comment (smoogipoo) [inline osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs:195]: condition inverted?

File: osu.Game/Beatmaps/WorkingBeatmap.cs [hunk 2]
```diff
@@ -353,4 +355,186 @@ public BeatmapLoadTimeoutException(BeatmapInfo beatmapInfo)
...
+        public static bool IsLoaded(this ITrack track)
+        {
+            if (track is Track t)
+                return t.IsLoaded;
+
+            if (track is LoggingTrack lt)
+                return lt.UnderlyingTrack.IsLoaded;
+
+            return false;
+        }
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/WorkingBeatmap.cs:384]: Is this required? `LoggingTrack` implements this as a redirect to `UnderlyingTrack.IsLoaded`.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs]: Fixed.

**Comment by peppy** [comment]
Have depoyed this via a branch, no need to keep it around in `master`.

**Pull request closed without merging**